### PR TITLE
Add support for IP field

### PR DESCRIPTION
**Pull request by evanxg852000**
Integrated Tantivy changes to support IP field types

### CODE REVIEW

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs [hunk 1]
```diff
@@ -92,6 +92,31 @@ impl Default for QuickwitNumericOptions {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, Debug)]
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs:95]:
```suggestion
#[derive(Clone, Debug, Serialize, Deserialize)]
```

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs [hunk 2]
```diff
@@ -958,6 +987,32 @@ mod tests {
...
+            "#,
+        )
+        .unwrap();
+        let entry_str = serde_json::to_value(&entry).unwrap();
+        assert_eq!(
+            entry_str,
+            serde_json::json!({
+                "name": "ip_address",
+                "description": "Client ip address",
+                "type":"ip",
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs:1008]:
```suggestion
                "type": "ip",
```

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs [hunk 3]
```diff
@@ -958,6 +987,32 @@ mod tests {
...
+            }
+            "#,
+        )
+        .unwrap();
+        let entry_str = serde_json::to_value(&entry).unwrap();
+        assert_eq!(
+            entry_str,
+            serde_json::json!({
+                "name": "ip_address",
+                "description": "Client ip address",
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs:1007]:
```suggestion
                "description": "Client IP address",
```

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs [hunk 4]
```diff
@@ -958,6 +987,32 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_parse_ip_addr_mapping() {
+        let entry = serde_json::from_str::<FieldMappingEntry>(
+            r#"
+            {
+                "name": "ip_address",
+                "description": "Client ip address",
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs:996]:
```suggestion
                "description": "Client IP address",
```

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs [hunk 5]
```diff
@@ -103,6 +108,16 @@ impl LeafType {
                     Err(format!("Expected bool value, got '{}'.", json_val))
                 }
             }
+            LeafType::IpAddr(_) => {
+                if let JsonValue::String(ip_address) = json_val {
+                    let ipv6_value = IpAddr::from_str(ip_address.as_str())
+                        .map_err(|err| format!("Failed to parse ip address '{ip_address}', {err}"))?
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs:114]:
```suggestion
                        .map_err(|err| format!("Failed to parse IP address `{ip_address}`: {err}"))?
```

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs (hunk 5 above)
Comment (evanxg852000) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs]: Regarding the quote, using `'` makes a bit of a mixture here so I will just use ``` in the other places.

**Pull request merged**